discord-bot/src/interactions/application_commands/matchmaking.rs

```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+
+                let lobby = MatchmakingLobbies::find()
+                    .filter(matchmaking_lobbies::Column::ChannelId.eq(IdWrapper::from(chan_id)))
+                    .one(self.utils.db_ref())
+                    .await?;
+
+                if lobby.is_none() {
+                    return Err(anyhow!(
+                        "You must run this command in a valid matchmaking thread."
+                    ));
+                }
+
+                // TODO: Refactor get resolved user from app command data to a function.
+                let resolved = data
+                    .command
+                    .resolved
+                    .ok_or_else(|| anyhow!("cannot get the resolved command user data"))?;
+                let opponent = resolved.users.values().next().ok_or_else(|| {
+                    anyhow!("cannot get the user specified in \"report-score\" command")
+                })?;
+
+                let response = self.utils.http_client.channel(chan_id).exec().await?;
+
+                if !response.status().is_success() {
+                    return Err(anyhow!("failed to get thread"));
+                }
+
+                let channel: Channel = response.model().await?;
+
+                let thread_members: Vec<ThreadMember> = self
+                    .utils
+                    .http_client
+                    .thread_members(channel.id)
+                    .exec()
+                    .await?
+                    .models()
+                    .await?;
+
+                let mut reporter_is_member: bool = false;
+                let mut opponent_is_member: bool = false;
+
+                for member in thread_members {
+                    let member_user_id = member.user_id.ok_or_else(|| {
+                        anyhow!("cannot get the user specified in \"report-score\" command")
+                    })?;
+                    if user.id == member_user_id {
+                        reporter_is_member = true;
+                    } else if opponent.id == member_user_id {
+                        opponent_is_member = true;
+                    } else if reporter_is_member && opponent_is_member {
+                        break;
+                    }
+                }
+
+                // TODO: Check if opponent is the bot, because that would be invalid.
+                if !reporter_is_member {
+                    return Err(anyhow!("user isn't part of the lobby"));
+                } else if !opponent_is_member {
+                    return Err(anyhow!("opponent isn't part of the lobby"));
+                }
+
+                let options: Vec<CommandDataOption>;
+                match subcommand.value.clone() {
+                    CommandOptionValue::SubCommand(x) => options = x,
+                    _ => options = Vec::<CommandDataOption>::new(),
+                }
+
+                let wins_option_value = options
+                    .get(1)
```

Comment:
I tried to re-order the values of the command, and it appears that it's always sent in the correct order. I don't foresee indexing the options like this to be an issue as I previously mentioned.